### ISSUE TITLE
[Bug] log noise prior is added twice in the MLL computation

### ISSUE THREAD

**Issue by LuisAugenstein** [cornellius-gp/gpytorch#2340]
# Noise prior is added twice when computing the ExactMarginalLogLikelihood

When I create a simple ExactGP Model with a noise prior I think the marginal log likelihood is computed wrong. 
In my opinion the MLL (or let's rather call it log posterior in this scenario as we have a noise prior) should be computed as follows

$\text{log } p(X, \sigma^2 \vert \boldsymbol{y}) = \text{log } p(\boldsymbol{y} \vert X, \sigma^2) + \text{log } p(\sigma^2)$

However, in my experiment it seems as if currently gpytorch adds the log noise prior twice instead of once.

$\text{log } p(X, \sigma^2 \vert \boldsymbol{y}) = \text{log } p(\boldsymbol{y} \vert X, \sigma^2) + 2 \ \text{log } p(\sigma^2)$

This is because the MarginalLogLikelihood module contains the likelihood module twice, i.e., directly via `self.likelihood` and indirectly through the model `self.model.likelihood`.
```python
class gpytorch.mlls.MarginalLogLikelihood(Module):
    def __init__(self, likelihood, model):
        self.likelihood = likelihood
        self.model = model
```

## To reproduce
```python
import torch
import gpytorch

torch.manual_seed(123)

class ExactGPModel(gpytorch.models.ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super(ExactGPModel, self).__init__(train_x, train_y, likelihood)
        self.mean_module = gpytorch.means.ConstantMean()
        self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel())

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)


N = 10
train_x = torch.randn(N)
train_y = torch.randn(N)
noise_prior = gpytorch.priors.NormalPrior(0, 1)
likelihood = gpytorch.likelihoods.GaussianLikelihood(noise_prior)
model = ExactGPModel(train_x, train_y, likelihood)

mll = gpytorch.mlls.ExactMarginalLogLikelihood(model.likelihood, model)
gp_prior = model(train_x)
marginal_log_likelihood = mll(gp_prior, train_y)

marginal_likelihood = likelihood(gp_prior)

marginal_log_likelihood_as_it_should_be = (marginal_likelihood.log_prob(
    train_y) + noise_prior.log_prob(likelihood.noise)) / N
assert torch.abs(marginal_log_likelihood - marginal_log_likelihood_as_it_should_be) > 0.001

marginal_log_likelihood_as_it_currently_is = (marginal_likelihood.log_prob(
    train_y) + 2*noise_prior.log_prob(likelihood.noise)) / N
assert torch.abs(marginal_log_likelihood - marginal_log_likelihood_as_it_currently_is) < 0.00001
```

## Current Behavior
Both assertions hold. The marginal_log_likelihood is computed by adding the log noise prior twice.

## Expected Behavior
Both assertions should fail. The `marginal_log_likelihood` should equal the `marginal_log_likelihood_as_it_should_be` variable as defined above, i.e., just adding the noise_prior once. 

## System information
- GPyTorch version 1.10
- PyTorch version 2.0.0
- Computer Os: Ubuntu 22.04.2 LTS

## Additional Note
I accidentally published this issue before I completed the description hence I temporarily closed the issue, finished the description and reopened it again.

**Comment by gpleiss** [comment]
@LuisAugenstein can you put up a PR to fix this?

**Comment by gpleiss** [comment]
> Should it be possible to use two different likelihoods?

No it shouldn't be.

@LuisAugenstein you're right that `MarginalLogLikelihood(model)` would be a simpler API, but we're not looking to make a breaking change at the moment.

> Maybe scale the priors log_prob then by 0.5 or so?

I think that's all you would need to do to fix this bug.

**Comment by LuisAugenstein** [comment]
> No it shouldn't be.
Ok great. I think in that case my first suggestion (to use `mll.named_priors()` instead of `model.named_priors()`) is more intuitive/readable than scaling both priors by 0.5. I will create the PR.

**Comment by gpleiss** [comment]
Closed by #2355
